File: Backend/src/main/java/com/family/modules/user/service/UserService.java
package com.family.modules.user.service;

import com.family.common.dto.PagingRequest;
import com.family.common.enums.RoleEnum;
import com.family.modules.user.dto.UserRequest;
import com.family.modules.user.entity.User;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.UUID;

public interface UserService {
    User getById(UUID id);
    List<User> getAll();
    Page<User> getPaged(PagingRequest request);
    User create(UserRequest request);
    User update(UUID id, UserRequest request);
    void delete(UUID id);
    User changeRole(UUID id, RoleEnum role);
}

File: Frontend/src/modules/admin/services/userService.js
import api from '@/services/api';

const userService = {
  getPaged: async (pageIndex = 1, pageSize = 10, keyword = '', role = 'ALL') => {
    const response = await api.post('/users/page', {
      pageIndex,
      pageSize,
      keyword,
      role,
    });
    return response.data;
  },
  changeRole: async (id, role) => {
    const response = await api.put(`/users/${id}/role`, null, { params: { role } });
    return response.data;
  },
  create: async (data) => {
    const response = await api.post('/users', data);
    return response.data;
  },
  update: async (id, data) => {
    const response = await api.put(`/users/${id}`, data);
    return response.data;
  },
  delete: async (id) => {
    const response = await api.delete(`/users/${id}`);
    return response.data;
  }
};

export default userService;
